src/main/java/Measurement.java

```diff
@@ -1,71 +1,90 @@
 import org.json.JSONArray;
 import org.json.JSONObject;
+
 import java.util.ArrayList;
 import java.util.Date;
 import java.util.List;
 import java.util.concurrent.locks.ReentrantReadWriteLock;
 
 //this class will hold jobs
-public class Measurement{
-   private static List<Job> activeJobs=new ArrayList<Job>();
-   private final static ReentrantReadWriteLock readWriteLock=new ReentrantReadWriteLock();
+public class Measurement {
+    private static List<Job> activeJobs;
+    private static ReentrantReadWriteLock readWriteLock;
 
-   public static boolean addMeasurement(JSONObject jobRequest){
-            acquireWriteLock();
-            if(jobRequest==null) return false;
-            //TODO there is need for error checking for valid jobRequest structure before addition
-            JSONObject jobDesc = jobRequest.getJSONObject("job_description");
-            Job jobTobeScheduled = new Job(jobDesc);
-            activeJobs.add(jobTobeScheduled);
-            releaseWriteLock();
-            return true;
-        }
-    public static JSONArray getActiveJobs() {
-            acquireReadLock();
-            JSONArray sentJobs=new JSONArray();
-            Date currentTime=new Date();
-            for(Job job:activeJobs) {
-                if(job.canStart(currentTime)&&!job.isResettable()&&!job.isRemovable()){
-                    sentJobs.put(job.getMeasurementDesc());
-                }
+    public static void init() {
+        activeJobs = new ArrayList<Job>();
+        readWriteLock = new ReentrantReadWriteLock();
+    }
+
+    public static boolean addMeasurement(JSONObject jobRequest) {
+        acquireWriteLock();
+        if (jobRequest == null) return false;
+        //TODO there is need for error checking for valid jobRequest structure before addition
+        System.out.println(jobRequest.toString());
+        JSONObject jobDesc = jobRequest.getJSONObject("job_description");
+        Job jobTobeScheduled = new Job(jobDesc);
+        activeJobs.add(jobTobeScheduled);
```

Comment:
think this and the check above are the only things that need to be in the critical section.